### PR TITLE
Ticket 7 - Adding Voting to Article

### PR DESCRIPTION
**Pull request by ChrisMaylor92**

### CODE REVIEW

**Review by mezzDavies — CHANGES_REQUESTED** [review]
This is getting there but I would have a read of [this section of the notes](https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-approach) and refactor it to act in a similar way. We want to optimistically render the votes when incremented and then if the patch was unsuccessful (eg a network error) then let the user know and take away that optimistic rendering. You only have the option for the user to upvote (which is fine) but there should be no reason for you make an API call to remove a vote if this isn't possible on the front end.

File: src/components/ArticleCard.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { patchArticle } from "../API"
+import { useState } from "react";
+import { Link } from "react-router-dom";
+
+export const ArticleCard = ({article}) => {
+    console.log(article)
```
Comment (nc-stephen-corke) [inline src/components/ArticleCard.jsx:6]: be sure to remove the log

File: src/API.js [hunk 2]
```diff
@@ -36,6 +36,19 @@ export const getAllArticlesPagnated = (pageNumber) => {
     })
 }
 
+export const patchArticle = (article_id, up) => {
+    if(up) {
```
Comment (nc-stephen-corke) [inline src/API.js:40]: the `up` logic is a little hard to read for me here. You could just pass the number that you want to change the votes by and then invoked the api function with `1` or `-1`

File: src/components/ArticleCard.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    const [hasVoted, setHasVoted] = useState(false)
+    const [voteCount, setVoteCount] = useState(article.votes)
+    const [err, setErr] = useState(null)
+
+    const handleVote = () => {
+        if (!hasVoted) {
```
Comment (nc-stephen-corke) [inline src/components/ArticleCard.jsx:12]: logic looks good here, nice work

**Review by nc-stephen-corke — COMMENTED** [review]
Refactor looks good - it's worth making the change suggested to the api function for readability. Merge when ready.

**Pull request merged**